The failed build was complaining about a `C:\Program` path from a generated file:
```
14:39:40   CMake Error at pybind11-2.5.0-stamp/download-pybind11-2.5.0.cmake:106 (message):
14:39:40     Syntax error in cmake code at
14:39:40       C:/ci/ws/build/pybind11_vendor/pybind11-2.5.0-prefix/src/pybind11-2.5.0-stamp/download-pybind11-2.5.0.cmake:108
14:39:40     when parsing string
14:39:40       C:\Program
14:39:40     Invalid character escape '\P'.
```

It's not clear where that path comes from. @hidmic suggested it looks like some paths were not converted correctly to CMake paths.

Suppressing warnings on Windows instead:
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=11693)](http://ci.ros2.org/job/ci_windows/11693/)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,9 @@ macro(build_pybind11)
   set(extra_cmake_args)
   # set(cmake_commands)
   if(DEFINED CMAKE_BUILD_TYPE)
-    list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
+    if(NOT WIN32)
+      list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
+    endif()
     # if(WIN32)
     #   list(APPEND extra_cmake_args "-Wno-dev")
     # endif()
@@ -60,7 +62,6 @@ macro(build_pybind11)
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install
       -DPYBIND11_TEST=OFF
-      -Wno-dev
       ${extra_cmake_args}
   )
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,9 +9,9 @@ macro(build_pybind11)
   # set(cmake_commands)
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
-    if(WIN32)
-      list(APPEND extra_cmake_args "-Wno-dev")
-    endif()
+    # if(WIN32)
+    #   list(APPEND extra_cmake_args "-Wno-dev")
+    # endif()
 
     # if(WIN32)
     #   build_command(_build_command CONFIGURATION ${CMAKE_BUILD_TYPE})
@@ -60,6 +60,7 @@ macro(build_pybind11)
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install
       -DPYBIND11_TEST=OFF
+      -Wno-dev
       ${extra_cmake_args}
   )
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,21 +6,10 @@ find_package(ament_cmake REQUIRED)
 
 macro(build_pybind11)
   set(extra_cmake_args)
-  # set(cmake_commands)
   if(DEFINED CMAKE_BUILD_TYPE)
     if(NOT WIN32)
       list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
     endif()
-    # if(WIN32)
-    #   list(APPEND extra_cmake_args "-Wno-dev")
-    # endif()
-
-    # if(WIN32)
-    #   build_command(_build_command CONFIGURATION ${CMAKE_BUILD_TYPE})
-    #   list(APPEND cmake_commands "BUILD_COMMAND ${_build_command}")
-    # else()
-    #   list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
-    # endif()
   endif()
   if(DEFINED CMAKE_TOOLCHAIN_FILE)
     list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
@@ -58,7 +47,6 @@ macro(build_pybind11)
     URL https://github.com/pybind/pybind11/archive/v2.5.0.tar.gz
     URL_MD5 1ad2c611378fb440e8550a7eb6b31b89
     TIMEOUT 600
-    # ${cmake_commands}
     CMAKE_ARGS
       -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_install
       -DPYBIND11_TEST=OFF

